pandas/tests/indexes/test_numeric.py

```diff
@@ -259,6 +259,12 @@ def test_astype(self):
         for dtype in ['M8[ns]', 'm8[ns]']:
             self.assertRaises(TypeError, lambda: i.astype(dtype))
 
+        # GH 13149
+        # a NaN astype int
```

Comment:
add tests for datetimelikes as well to cover the bases; I think your change might be changing behavior there as well (which is not correct)